src/mozanalysis/frequentist_stats/sample_size.py

```diff
@@ -0,0 +1,309 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from scipy.stats import norm
+from math import pi
+from statsmodels.stats.power import tt_ind_solve_power, zt_ind_solve_power
+from mozanalysis.bq import BigQueryContext
+from mozanalysis.experiment import TimeSeriesResult
+from statsmodels.stats.proportion import samplesize_proportions_2indep_onetail
+import numpy as np
+
+
+def difference_of_proportions_sample_size_calc(
+    df,
+    metrics_list,
+    effect_size=.01,
+    alpha=.05,
+    power=.90,
+    outlier_percentile=99.5
+):
+
+    """
+    Perform sample size calculation for an experiment to test for a
+    difference in proportions.
+
+    Args:
+        df: A pandas DataFrame of queried historical data.
+        metrics_list (list of mozanalysis.metrics.Metric): List of metrics
+            used to construct the results df from HistoricalTarget. The names
+            of these metrics are used to return results for sample size
+            calculation for each
+        effect_size (float, default .01): Difference in proportion for the
+            minimum detectable effect:
+                effect_size = p(event under alt) - p(event under null)
+        alpha (float, default .05): Significance level for the experiment.
+        power (float, default .90): Probability of detecting an effect,
+            when a significant effect exists.
+        outlier_percentile(float, default .995): Percentile at which to trim
+            each columns.
+
+    Returns a sorted dictionary:
+        Keys in the dictionary are the metrics column names from the DataFrame; values
+        are the required sample size to achieve the desired power for that metric.
+        The dictionary is sorted by required sample size, in descending order.
+    """
+
+    def _get_sample_size_col(col):
+
+        p = np.percentile(df[col], q=[outlier_percentile])[0]
+        mean = df.loc[df[col] <= p, col].mean()
+        p2 = mean + effect_size
+
+        return samplesize_proportions_2indep_onetail(
+            diff=effect_size,
+            prop2=p2,
+            power=power,
+            ratio=1,
+            alpha=alpha,
+            value=0
+        )
+
+    metric_names = [m.name for m in metrics_list]
+    results = {}
+    for col in metric_names:
+        sample_size = _get_sample_size_col(col)
+        pop_percent = 100.*(sample_size/len(df))
+        results[col] = {
+            "sample_size": sample_size,
+            "population_percent": pop_percent
```

Comment:
```suggestion
            "population_percent": pop_percent,
             "population": len(df)
```
Could we include the population as well? It's helpful to have this number as a sanity check.

Reply:
Added!

Reply:
thanks!